ramips: add support for Linksys E5600

digislayer opened the pull request:
This submission relied heavily on the work of Linksys EA7300 v1/ v2.

Specifications:

* SoC: MediaTek MT7621A (880 MHz 2c/4t)
* RAM: 128M DDR3-1600
* Flash: 128M NAND
* Eth: MediaTek MT7621A (10/100/1000 Mbps x5)
* Radio: MT7603E/MT7613BE (2.4 GHz & 5 GHz)
* Antennae: 2 internal fixed in the casing and 2 on the PCB
* LEDs: Blue (x4 Ethernet)
  Blue+Orange (x2 Power + WPS and Internet)
* Buttons: Reset (x1)
  WPS (x1)

Installation:

Flash factory image through GUI.

This device has 2 partitions for the firmware called firmware and
alt_firmware. To successfully flash and boot the device, the device
should have been running from alt_firmware partition. To get the device
booted through alt_firmware partition, download the OEM firmware from
Linksys website and upgrade the firmware from web GUI. Once this is done,
flash the OpenWrt Factory firmware from web GUI.

Reverting to factory firmware:

1. Boot to 'alt_firmware'(where stock firmware resides) by doing one of
   the following:
   Press the "wps" button as soon as power LED turns on when booting.
   (OR) Hard-reset the router consecutively three times to force it to
   boot from 'alt_firmware'.
2. To remove any traces of OpenWRT from your router simply flash the OEM
   image at this point.

Signed-off-by: Aashish Kulkarni <aashishkul@gmail.com>

[fix hanging indents and wrap to 74 characters per line,
 add kmod-mt7663-firmware-sta package for 5GHz STA mode to work,
 remove sysupgrade.bin and concatenate IMAGES instead in mt7621.mk,
 set default-state "on" for power LED]
Signed-off-by: Sannihith Kinnera <digislayer@protonmail.com>

aashishkul committed:
ramips: add support for Linksys E5600

This submission relied heavily on the work of Linksys EA7300 v1/ v2.

Specifications:

* SoC: MediaTek MT7621A (880 MHz 2c/4t)
* RAM: 128M DDR3-1600
* Flash: 128M NAND
* Eth: MediaTek MT7621A (10/100/1000 Mbps x5)
* Radio: MT7603E/MT7613BE (2.4 GHz & 5 GHz)
* Antennae: 2 internal fixed in the casing and 2 on the PCB
* LEDs: Blue (x4 Ethernet)
  Blue+Orange (x2 Power + WPS and Internet)
* Buttons: Reset (x1)
  WPS (x1)

Installation:

Flash factory image through GUI.

This device has 2 partitions for the firmware called firmware and
alt_firmware. To successfully flash and boot the device, the device
should have been running from alt_firmware partition. To get the device
booted through alt_firmware partition, download the OEM firmware from
Linksys website and upgrade the firmware from web GUI. Once this is done,
flash the OpenWrt Factory firmware from web GUI.

Reverting to factory firmware:

1. Boot to 'alt_firmware'(where stock firmware resides) by doing one of
   the following:
   Press the "wps" button as soon as power LED turns on when booting.
   (OR) Hard-reset the router consecutively three times to force it to
   boot from 'alt_firmware'.
2. To remove any traces of OpenWRT from your router simply flash the OEM
   image at this point.

Signed-off-by: Aashish Kulkarni <aashishkul@gmail.com>

[fix hanging indents and wrap to 74 characters per line,
 add kmod-mt7663-firmware-sta package for 5GHz STA mode to work,
 remove sysupgrade.bin and concatenate IMAGES instead in mt7621.mk,
 set default-state "on" for power LED]
Signed-off-by: Sannihith Kinnera <digislayer@protonmail.com>

digislayer commented:
@adschm : Creating a new PR per your [advice](https://github.com/openwrt/openwrt/pull/3958#issuecomment-840799464).

This PR should be replacing https://github.com/openwrt/openwrt/pull/3958

adschm commented:
So, let's do what we always do with new PRs from old PRs:

"This has formal issues ..."

Please format your Signed-off-by appropriately, i.e. take the original commit, change it, and then annotate your changes before your own Signed-off-by in a very brief way... No multi-SoB etc.

digislayer commented:
> So, let's do what we always do with new PRs from old PRs:
> 
> "This has formal issues ..."
> 
> Please format your Signed-off-by appropriately, i.e. take the original commit, change it, and then annotate your changes before your own Signed-off-by in a very brief way... No multi-SoB etc.

Sorry, I'm new to this.. and not confident I understood your comment correctly.
Just to be sure we're on the same page:

1. Is it okay if I just edit my commit message, so that it includes only my SoB? Or, do I still include TWO SoB(one by original author @aashishkul , another by me @digislayer)?
2. By Annotating my changes - if there are no functional changes, do I still include additional annotations? Ex: If I just changed from "IMAGES :=" to "IMAGES +=" in the DTS file, modified included device packages, resolved merge conflicts.
3. Is it okay to have at least 2 commits, squashed together(one from original author, next from me)? If yes, do I include the newer commit ID(from me) in the squashed commit-message?

Borromini commented:
As for the sign-off: just one SoB like Adrian says (yours), you can still say in your commit message whose work this PR is based on.

adschm commented:
What you need is:

```
Signed-off-by: Aashish Kulkarni <aashishkul@gmail.com>
[change 1, change 2, change 3, change 4]
Signed-off-by: Sannihith Kinnera <digislayer@protonmail.com>
```

adschm commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -762,6 +766,25 @@ define Device/lenovo_newifi-d1
+  IMAGES += factory.bin
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata | check-size
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | \
+  append-ubi | check-size | gemtek-trailer
```
tab for hanging indent.

digislayer replied:
This has been fixed, please refer to my comment further down.

adschm commented:
Apart from formal stuff, there seem to be no big issues left.

digislayer commented on target/linux/ramips/dts/mt7621_linksys_e5600.dts:
```diff
@@ -0,0 +1,181 @@
+
+		led_power: power {
+			label = "blue:power";
+			gpios = <&gpio 10 GPIO_ACTIVE_HIGH>;
```
Yes. I just tested with "GPIO_ACTIVE_LOW", and the LED stayed "off" even after powering on.
Other devices like Linksys_EA7xxx(based on which this patch has been developed) also have similar behavior.

adschm replied:
What I meant was that you should put "state-default = on" here, so the led gets lid earlier in the boot process...

digislayer replied:
This device shares the power LED with other functions - led-boot, led-failsafe, led-running, led-upgrade.
Is it still safe to have "default-state = on"?

I have just built and tested with "default-state = on", and it seems to be working fine. Committing this fix for now, but please suggest about the safety of this.

digislayer commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -22,6 +22,10 @@ define Build/elecom-wrc-gs-factory
 endef
 
+define Build/gemtek-trailer
+	printf "%s%08X" ".GEMTEK." "$$(cksum $@ | cut -d ' ' -f1)" >> $@
```
Hanging indent has been fixed.

digislayer commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -762,6 +766,25 @@ define Device/lenovo_newifi-d1
+  DEVICE_VENDOR := Linksys
+  DEVICE_MODEL := E5600
+  DEVICE_PACKAGES := kmod-mt7603 kmod-mt7615e kmod-mt7663-firmware-ap \
+	kmod-mt7663-firmware-sta uboot-envtools
```
Wrapped to 74 characters max, and fixed hanging indent with single tab.

digislayer commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -762,6 +766,25 @@ define Device/lenovo_newifi-d1
+  IMAGES += factory.bin
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata | check-size
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | \
+	append-ubi | check-size | gemtek-trailer
```
Hanging indent has been fixed with single tab.

digislayer commented:
@adschm, I hope the SoB lines are per your advice now. I have fixed the indentation and wrapping issues too.
Please check and let me know if anything more is needed.

adschm commented:
I don't understand your organization of the authorship:

> Factory image checksum calculation optimization for Linksys E5600
> 
> Dropping kmod-usb3 package and disabled in the dts as nothing is connected to USB.
> 
> Signed-off-by: Aashish Kulkarni <aashishkul@gmail.com>

If the changes above are aashishkul's work why mention them separately?

> ramips: Fixing Linksys e5600 image config in mt7621.mk
> 
> ramips: Fixing Linksys e5600 image config in mt7621.mk
> 
> ramips: fix formatting issues for Linksys e5600
> 
> Signed-off-by: Sannihith Kinnera <digislayer@protonmail.com>

Why do name the same (generic) change twice? And please use the syntax I already provided in my earlier comment:

```
Signed-off-by: Aashish Kulkarni <aashishkul@gmail.com>
[fix 1, fix 2, fix 3]
Signed-off-by: Sannihith Kinnera <digislayer@protonmail.com>
```

Apart from that, please be specific but still brief. Use imperativ form. "fixing e5600 image config" does not tell me much, but "drop unneeded SUPPORTED_DEVICES" (as an example) tells me everything ...

digislayer commented:
> I don't understand your organization of the authorship:
> 
> > Factory image checksum calculation optimization for Linksys E5600
> > Dropping kmod-usb3 package and disabled in the dts as nothing is connected to USB.
> > Signed-off-by: Aashish Kulkarni [aashishkul@gmail.com](mailto:aashishkul@gmail.com)
> 
> If the changes above are aashishkul's work why mention them separately?
> 
> > ramips: Fixing Linksys e5600 image config in mt7621.mk
> > ramips: Fixing Linksys e5600 image config in mt7621.mk
> > ramips: fix formatting issues for Linksys e5600
> > Signed-off-by: Sannihith Kinnera [digislayer@protonmail.com](mailto:digislayer@protonmail.com)
> 
> Why name the same (generic) change twice? And please use the syntax I already provided in my earlier comment:
> 
> ```
> Signed-off-by: Aashish Kulkarni <aashishkul@gmail.com>
> [fix 1, fix 2, fix 3]
> Signed-off-by: Sannihith Kinnera <digislayer@protonmail.com>
> ```
> 
> Apart from that, please be specific but still brief. Use imperative form. "fixing e5600 image config" does not tell me much, but "drop unneeded SUPPORTED_DEVICES" (as an example) tells me everything ...

I thought I need to mention changes commit by commit. Hence, all commit messages have been listed(Git's default when squashing changes).
If that is not needed and changes can be summarised, I will redo the authorship.

digislayer commented:
@adschm, Fixed authorship as following:
```
Signed-off-by: Aashish Kulkarni <aashishkul@gmail.com>

[fix hanging indents and wrap to 74 characters per line,
 add kmod-mt7663-firmware-sta package for 5GHz STA mode to work,
 remove sysupgrade.bin and concatenate IMAGES instead in mt7621.mk,
 set default-state "on" for power LED]
Signed-off-by: Sannihith Kinnera <digislayer@protonmail.com>
```
Had to split my changes into multiple lines in hopes of meet wrapping requirements. Please let me know how this looks.
Also modified the entire commit message to have maximum of 75 characters per line.

adschm commented:
> I thought I need to mention changes commit by commit. Hence, all commit messages have been listed(Git's default when squashing changes).

But this is only one commit. You just list everything you changed based on what the original author did.

adschm commented:
Yes, the new format catches the idea (have not checked the content yet).

digislayer commented:
> Yes, the new format catches the idea (have not checked the content yet).

Looking forward to your feedback once you're able to check the content.

digislayer commented:
@adschm, just checking to see if you think any further changes are needed here...
Are there any next steps I need to take?

adschm commented:
Okay, I'll merge it. Thanks!
